cli/azd/cmd/auth_login.go

```diff
@@ -254,81 +257,103 @@ func (la *loginAction) Run(ctx context.Context) (*actions.ActionResult, error) {
 			"Next time use `azd auth login`.")
 	}
 
-	if !la.flags.onlyCheckStatus {
-		if err := la.accountSubManager.ClearSubscriptions(ctx); err != nil {
-			log.Printf("failed clearing subscriptions: %v", err)
+	if la.flags.onlyCheckStatus {
+		// In check status mode, we always print the final status to stdout.
+		// We print any non-setup related errors to stderr.
+		// We always return a zero exit code.
+		token, err := la.verifyLoggedIn(ctx)
+		var loginExpiryError *auth.ReLoginRequiredError
+		if err != nil &&
+			!errors.Is(err, auth.ErrNoCurrentUser) &&
+			!errors.As(err, &loginExpiryError) {
+			fmt.Fprintln(la.console.Handles().Stderr, err.Error())
 		}
 
-		if err := la.login(ctx); err != nil {
-			return nil, err
+		if la.formatter.Kind() == output.NoneFormat {
+			if err != nil {
+				fmt.Fprintln(la.console.Handles().Stdout, "Not logged in, run `azd auth login` to login to Azure.")
+			} else {
+				fmt.Fprintln(la.console.Handles().Stdout, cLoginSuccessMessage)
+			}
+
+			return nil, nil
+		} else {
+			res := contracts.LoginResult{}
+			if err != nil {
```

Comment:
Can you try to avoid the 3rd level of `if-else`  
  1) if la.flags.onlyCheckStatus
  2) if la.formatter.Kind() == output.NoneFormat
  3) if err != nil

No one will die if we just keep adding nested code, but it is typically a `code smell` there.  I would try to make it simpler

Reply:
The code smell is that the formatter is a pattern that needs some improvements IMO.

Reply:
Simplified the behavior here to read better. Thanks for the suggestion.